pages/home.py

```diff
@@ -447,3 +776,138 @@ def render_overview_total_figure(data, country):
     ) * 100
 
     return functional_figure(func_df), functional_narrative(func_df)
+
+
+@callback(
+    Output("year-slider", "marks"),
+    Output("year-slider", "value"),
+    Output("year-slider", "min"),
+    Output("year-slider", "max"),
+    Input("stored-basic-country-data", "data"),
+    Input("country-select", "value"),
+)
+def update_year_range(data, country):
+    data = data["basic_country_info"]
+    expenditure_years = data[country].get("expenditure_years", [])
+    poverty_years = data[country].get("poverty_years", [])
+    common_years = set(poverty_years).intersection(set(expenditure_years))
+    min_year, max_year = expenditure_years[0], expenditure_years[-1]
```

Comment:
the expenditure_years could be unavailable. 
let's add a handling for this e.g. (Uruguay)
![image](https://github.com/user-attachments/assets/4647993e-e044-4046-8be4-9ae151bed23c)

Reply:
We could add a simple try catch statement for a general error handling.

Reply:
I added an explicit if/else condition instead of try/except. If the expenditure_years list is empty then I now do not show the year slider as well.

Reply:
sounds good! 
Could you also add a error handling for a general case? 
e.g. 

```python
try:
   lines 808 - 828
except:
   return "There was a problem processing the visualization"
```

This is because an error can occur for any other reasons. (e.g. we accidentally change the column names in our databricks and the column "poverty_years" could be renamed to "poverty_year".(just an example) This will result in a reference error. It would be better to contain the error within a plot than to allow it to affect the whole app.

Reply:
I see. The data here is being used to construct the slider itself and not the plots. The input is from stored-basic-country-data where I process and store some basic data from expenditure and poverty side so that these computations need  not be done each time the slider is updated. The poverty_years etc is my own nomenclature and not from the databricks columns.

I can add the try/except blocks in the render_subnational_spending_figures (and the poverty plots) instead where the outputs are functions. Would this be better, or am I misunderstanding something?

Reply:
Let's put the error handling in both places, the slider is a separate component from the figure. So the error which happens in the slider will not be caught even if we add the error handling within the figure.
The edge case which i mentioned above is just an example. I just wanted to suggest that things can go wrong in different ways, and that is why we need a general error handling. (this was not to suggest an error handling for that specific edge case)

Reply:
Sounds good! I was finding it a bit awkward to send the error text since currently the call back returns the parameters of the slider. I'll find a way around that and make the changes.

Reply:
Added a try/except block to output a disabled slider when an unexpected error is encountered. Additionally added an empty plot with annotation "An error was encountered when producing this figure" for the functions producing plots.